integration-tests/integration/dpsuser.search-with-filter.spec.js

```diff
@@ -0,0 +1,234 @@
+const parse = require('csv-parse')
+const { goToSearchWithFilterPage } = require('../support/dpsuser.helpers')
+const UserPage = require('../pages/userPage')
+
+context('DPS search with filter user functionality', () => {
+  before(() => {
+    cy.clearCookies()
+  })
+
+  beforeEach(() => {
+    cy.task('reset')
+  })
+
+  it('Should show filter', () => {
+    const searchWithFilter = goToSearchWithFilterPage({})
+    searchWithFilter.filter().should('exist')
+  })
+
+  it('can add and remove user filter', () => {
+    const searchWithFilter = goToSearchWithFilterPage({})
+
+    searchWithFilter.filterUser('Andy')
+    searchWithFilter.filterWithTag('Andy').should('exist')
+    searchWithFilter.userFilterInput().should('have.value', 'Andy')
+
+    searchWithFilter.filterWithTag('Andy').click()
+    searchWithFilter.filterWithTag('Andy').should('not.exist')
+    searchWithFilter.userFilterInput().should('have.value', '')
+  })
+
+  it('can change default ALL status to active or inactive only', () => {
+    const searchWithFilter = goToSearchWithFilterPage({})
+    searchWithFilter.statusFilterRadioButton('All').should('be.checked')
+
+    searchWithFilter.filterStatus('Inactive')
+    searchWithFilter.statusFilterRadioButton('Inactive').should('be.checked')
+    searchWithFilter.filterWithTag('Inactive').should('exist')
+
+    searchWithFilter.filterStatus('Active')
+    searchWithFilter.statusFilterRadioButton('Active').should('be.checked')
+    searchWithFilter.filterWithTag('Active').should('exist')
+
+    searchWithFilter.filterStatus('All')
+    searchWithFilter.statusFilterRadioButton('All').should('be.checked')
+    searchWithFilter.filterWithTag('All').should('not.exist')
+  })
+
+  it('can add and remove a single caseload filter', () => {
+    const searchWithFilter = goToSearchWithFilterPage({})
+
+    searchWithFilter.filterCaseload('Moorland')
+    searchWithFilter.filterWithTag('Moorland').should('exist')
+    searchWithFilter.filterWithTag('Active caseload only').should('exist')
+    searchWithFilter.activeCaseloadFilterRadioButton('Yes').should('be.checked')
+
+    searchWithFilter.filterWithTag('Active caseload only').click()
+    searchWithFilter.filterWithTag('Active caseload only').should('not.exist')
+    searchWithFilter.activeCaseloadFilterRadioButton('Yes').should('not.be.checked')
+
+    searchWithFilter.filterWithTag('Moorland').click()
+    searchWithFilter.filterWithTag('Moorland').should('not.exist')
+    searchWithFilter.activeCaseloadFilterRadioButton('Yes').should('be.checked')
+  })
+  it('can not see caseload filter when not an admin', () => {
+    const searchWithFilter = goToSearchWithFilterPage({ isAdmin: false })
```

Comment:
This got me thinking - what if a non-admin user edited the URL to add a caseload filter? Is this prevented?

Reply:
It would have no affect - for non-admin users the caseload isn't even supplied to API. It users a different backend API that just receives the logged in user to determine the caseload